rotkehlchen/chain/ethereum/modules/makerdao/decoder.py

```diff
@@ -561,31 +561,31 @@ def decode_saidai_migration(
 
     def addresses_to_decoders(self) -> Dict[ChecksumEvmAddress, Tuple[Any, ...]]:
         return {
-            MAKERDAO_BAT_A_JOIN.address: (self.decode_makerdao_vault_action, A_BAT, 'BAT-A'),
-            MAKERDAO_ETH_A_JOIN.address: (self.decode_makerdao_vault_action, A_ETH, 'ETH-A'),
-            MAKERDAO_ETH_B_JOIN.address: (self.decode_makerdao_vault_action, A_ETH, 'ETH-B'),
-            MAKERDAO_ETH_C_JOIN.address: (self.decode_makerdao_vault_action, A_ETH, 'ETH-C'),
-            MAKERDAO_KNC_A_JOIN.address: (self.decode_makerdao_vault_action, A_KNC, 'KNC-A'),
-            MAKERDAO_TUSD_A_JOIN.address: (self.decode_makerdao_vault_action, A_TUSD, 'TUSD-A'),
-            MAKERDAO_USDC_A_JOIN.address: (self.decode_makerdao_vault_action, A_USDC, 'USDC-A'),
-            MAKERDAO_USDC_B_JOIN.address: (self.decode_makerdao_vault_action, A_USDC, 'USDC-B'),
-            MAKERDAO_USDT_A_JOIN.address: (self.decode_makerdao_vault_action, A_USDT, 'USDT-A'),
-            MAKERDAO_WBTC_A_JOIN.address: (self.decode_makerdao_vault_action, A_WBTC, 'WBTC-A'),
-            MAKERDAO_WBTC_B_JOIN.address: (self.decode_makerdao_vault_action, A_WBTC, 'WBTC-B'),
-            MAKERDAO_WBTC_C_JOIN.address: (self.decode_makerdao_vault_action, A_WBTC, 'WBTC-C'),
-            MAKERDAO_ZRX_A_JOIN.address: (self.decode_makerdao_vault_action, A_ZRX, 'ZRX-A'),
-            MAKERDAO_MANA_A_JOIN.address: (self.decode_makerdao_vault_action, A_MANA, 'MANA-A'),
-            MAKERDAO_PAXUSD_A_JOIN.address: (self.decode_makerdao_vault_action, A_PAX, 'PAXUSD-A'),
-            MAKERDAO_COMP_A_JOIN.address: (self.decode_makerdao_vault_action, A_COMP, 'COMP-A'),
-            MAKERDAO_LRC_A_JOIN.address: (self.decode_makerdao_vault_action, A_LRC, 'LRC-A'),
-            MAKERDAO_LINK_A_JOIN.address: (self.decode_makerdao_vault_action, A_LINK, 'LINK-A'),
-            MAKERDAO_BAL_A_JOIN.address: (self.decode_makerdao_vault_action, A_BAL, 'BAL-A'),
-            MAKERDAO_YFI_A_JOIN.address: (self.decode_makerdao_vault_action, A_YFI, 'YFI-A'),
-            MAKERDAO_GUSD_A_JOIN.address: (self.decode_makerdao_vault_action, A_GUSD, 'GUSD-A'),
-            MAKERDAO_UNI_A_JOIN.address: (self.decode_makerdao_vault_action, A_UNI, 'UNI-A'),
-            MAKERDAO_RENBTC_A_JOIN.address: (self.decode_makerdao_vault_action, A_RENBTC, 'RENBTC-A'),  # noqa: E501
-            MAKERDAO_AAVE_A_JOIN.address: (self.decode_makerdao_vault_action, A_AAVE, 'AAVE-A'),
-            MAKERDAO_MATIC_A_JOIN.address: (self.decode_makerdao_vault_action, A_MATIC, 'MATIC-A'),
+            MAKERDAO_BAT_A_JOIN.address: (self.decode_makerdao_vault_action, A_BAT.resolve_to_crypto_asset(), 'BAT-A'),  # noqa: E501
```

Comment:
This is fine here only because `addresses_to_decoders` is called once in the lifetime of the decoder.

If this was called multiple times you would have needed to move them to the constructor and reuse them already resolved here.

Not a great approach and it's definitely something we need to improve on,